pkg/osm/utils_test.go

```diff
@@ -0,0 +1,160 @@
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osmcontrollerdeployment2",
+						Namespace: "osmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+			},
+			namespace: "osmnamespace",
+			pass:      false,
+		},
+		{
+			name: "there is one deployment in the controller namespace",
+			deployments: []*v1.Deployment{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osmcontrollerdeployment",
+						Namespace: "osmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+			},
+			namespace: "osmnamespace",
+			pass:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert := tassert.New(t)
+			objs := make([]runtime.Object, len(test.deployments))
+			for i := range test.deployments {
+				objs[i] = test.deployments[i]
+			}
+			k8s := fake.NewSimpleClientset(objs...)
+			deployment, err := GetOSMControllerDeployment(k8s, test.namespace)
+			if test.pass {
+				assert.NoError(err)
+				assert.NotNil(deployment)
+				assert.Equal(test.deployments[0].Name, deployment.Name)
+			} else {
+				assert.Error(err)
+			}
+		})
+	}
+}
+
+func TestFormatReleaseVersion(t *testing.T) {
+	tests := []struct {
+		name                      string
+		version                   string
+		expectedMajorMinorVersion string
+		pass                      bool
+	}{
+		{
+			name:                      "major, minor and patch version",
+			version:                   "v0.9.0",
+			expectedMajorMinorVersion: "v0.9",
+			pass:                      true,
+		},
+		{
+			name:                      "major and minor version",
+			version:                   "v0.8",
+			expectedMajorMinorVersion: "v0.8",
+			pass:                      true,
+		},
+		{
+			name:                      "release cut version",
+			version:                   "v0.8.1-rc.1",
+			expectedMajorMinorVersion: "v0.8",
+			pass:                      true,
+		},
+		{
+			name:                      "major version",
+			version:                   "v1",
+			expectedMajorMinorVersion: "",
+			pass:                      false,
+		},
+		{
+			name:                      "empty version",
+			version:                   "",
+			expectedMajorMinorVersion: "",
+			pass:                      false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert := tassert.New(t)
+			majorMinorVersion, err := FormatReleaseVersion(test.version)
+			if test.pass {
```

Comment:
Similar point, to change the `pass` field to `expErr` so that we can have the check as
`assert.Equal(test.expErr, err != nil)`